### DISCUSSION TITLE
Can Premake generate .csproj?

### DISCUSSION THREAD

**Discussion by ghost** [premake/premake-core#2306] [General]
I want to decouple from Microsoft's toolchain so I decided to use bflat (https://github.com/bflattened/bflat). But bflat itself is only a compiler like gcc, it doesn't handle many things like the all-in-one `dotnet` tool. If I want a .csproj to have code completion via the OmniSharp language server, or I need to use bflata (https://github.com/xiaoyuvax/bflata) when my project is more complex, I need to use the `dotnet` tool to create the .csproj first. I don't want to do so. Can Premake help me generate .csproj?

I tried to use Sharpmake (https://github.com/ubisoft/Sharpmake) but I can't wrap my mind around it. Using C# as a DSL language is a very bad idea IMO.

**Comment by nickclark2016** [comment]
Premake has C# support, yes.
